migration/migrate_sqlite_mysql.py

```diff
@@ -0,0 +1,77 @@
+"""
+    Use this file for migrating your data from SQLite to MySQL. It only works for version >=2.0.0 of the Flask
+    Monitoring Dashboard. For migrating your data from version 1.x.x to 2.y.y, use the script "migrate_v1_to_v2.py".
+
+    Before running the script, make sure to change the SQLITE_URL and MYSQL_URL variables. Refer to
+    http://docs.sqlalchemy.org/en/latest/core/engines.html on how to configure them. Also, make sure MySQL Server
+    is running on your machine.
+"""
+import timeit
+
+from contextlib import contextmanager
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+
+from flask_monitoringdashboard.database import Base, Endpoint, Request, Outlier, CodeLine, StackLine, Test, \
+    TestResult, TestEndpoint
+
+SQLITE_URL = 'sqlite:////Users/user/Flask-MonitoringDashboard/fmd_copy.db'
+MYSQL_URL = 'mysql+pymysql://user:password@localhost:3306/db2'
+ENTITIES = [Endpoint, Request, Outlier, CodeLine, StackLine, Test, TestResult, TestEndpoint]
+
+
+def create_db_session(db_url, clean=False):
+    engine = create_engine(db_url)
+    if clean:
+        Base.metadata.drop_all(engine)
+        Base.metadata.create_all(engine)
+    Base.metadata.bind = engine
+    return sessionmaker(bind=engine)
+
+
+@contextmanager
+def session_scope(db_session):
+    session = db_session()
+    try:
+        yield session
+        session.commit()
+    except Exception:
+        session.rollback()
+        raise
+    finally:
+        session.close()
+
+
+def get_old_instances(db_session_old, entity_class):
+    with session_scope(db_session_old) as db_session:
+        old_instances = db_session.query(entity_class).all()
+        db_session.expunge_all()
+    return old_instances
+
+
+def migrate_entity(db_session_old, db_session_new, entity_class):
+    old_instances = get_old_instances(db_session_old, entity_class)
+    new_instances = []
+    with session_scope(db_session_new) as db_session:
+        for old_instance in old_instances:
+            del old_instance.__dict__['_sa_instance_state']
+            new_instances.append(entity_class(**old_instance.__dict__))
+        db_session.bulk_save_objects(new_instances)
+
+
+def main():
+    db_session_old = create_db_session(SQLITE_URL)
+    db_session_new = create_db_session(MYSQL_URL, clean=True)
+    t0 = timeit.default_timer()
+    t_previous = t0
+    for entity_class in ENTITIES:
+        migrate_entity(db_session_old, db_session_new, entity_class)
+        t_now = timeit.default_timer()
+        print("Moving %s took %f seconds" % (entity_class.__name__, t_now - t_previous))
+        t_previous = t_now
+    print("Total time was %f seconds" % (t_previous - t0))
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
got an `ModuleNotFoundError: No module named 'pymysql'`

Is the `mysql+pymysql` really needed? 

I seem to be able to run the script with only `mysql`... but it's waiting A LOT for for request

Reply:
Oki, that is nothing compared with StackLine :)

Reply:
hmm. the thing seems frozen at StackLine since more than 20 min (my subjective perception)

```
Moving Request took 78.195023 seconds
/usr/local/lib/python3.6/site-packages/mysqlclient-1.3.10-py3.6-linux-x86_64.egg/MySQLdb/cursors.py:313: Warning: (1265, "Data truncated for column 'stacktrace' at row 1")
  rows += self.execute(sql + postfix)
/usr/local/lib/python3.6/site-packages/mysqlclient-1.3.10-py3.6-linux-x86_64.egg/MySQLdb/cursors.py:318: Warning: (1265, "Data truncated for column 'stacktrace' at row 1")
  rows += self.execute(sql + postfix)
/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/default.py:508: Warning: (1265, "Data truncated for column 'stacktrace' at row 1")
  cursor.execute(statement, parameters)

Moving Outlier took 47.479775 seconds
Moving CodeLine took 1.444501 seconds
```

Reply:
I'm actually getting an error before that. The Outlier with id 18 has a stacktrace of 69k characters and the max limit for MySQL TEXT is 65k characters. I'm surprised it worked for you.

Reply:
Ok, I see what you mean about the StackLine. The data is huge and my computer also froze. I will implement a mechanism that moves the data into chunks of fixed size.